### ISSUE TITLE
SVM for regression: why? versus linear regression?

### ISSUE THREAD

**Issue by JPFrancoia** [ageron/handson-ml#312]
Hi,

I'm reading the chapter on the SVMs at the moment, and it's not clear to me why I would use one for regression. When used for regression, and if I understand well, the SVM will try to put as much instances as possible in the margins. The outliers seem to have an important influence on these margins.

What is the difference with linear regression, which tries to minimize the MSE?

And why would one choose SVM vs linear regression? I'm trying to understand what are the pros and cons of each algorithm, so an example would greatly help me to understand.

Best (and thanks for this book, it's amazing!)

**Comment by ageron** [comment]
Thanks @JPFrancoia ,
Linear Regression and SVM do not have the same objective or training algorithm, so they may perform differently on different datasets.  Sometimes LR will be better, sometimes it will be SVM: you can train one of each and evaluate them both on the validation set, to see which one works best in your case.  That said, on a linear problem, where the noise is Gaussian, they will often produce more or less the same linear model.  People generally use SVMs with a kernel (such as the RBF kernel), which makes it possible to tackle non-linear problems. That's one thing that LR cannot do.  Conversely, using Gradient Descent, it is possible to scale LR to very large dataset, but an kernelized SVM cannot scale to large datasets (it shines for fairly small datasets).
Hope this helps,
Aurélien

**Comment by JPFrancoia** [comment]
Thanks @ageron for this explanation. Yes in my head I was comparing the two on a linear problem. They seemed to do more or less the same thing but differently.

Thanks!
